syntax = "proto3";

package member_service;

import "google/protobuf/timestamp.proto";

// The Member service definition.
service MemberService {
  // A CreateMember RPC that takes a request and returns creation result.
  rpc CreateMember(CreateMemberRequest) returns (CreateMemberResponse) {}

   // A UpdateMember RPC that takes a request and returns updation result.
  rpc UpdateMember(UpdateMemberRequest) returns (UpdateMemberResponse) {}
}

// request for CreateMember.
message CreateMemberRequest {
  string name = 1;
  string phone = 2;
  google.protobuf.Timestamp created_at = 3;
}

// The response message definition, return status of creation.
message CreateMemberResponse {
 bool is_member_created = 1;
}

// request message of UpdateMember.
message UpdateMemberRequest {
  string id = 1;
  string name = 2;
  string phone = 3;
  google.protobuf.Timestamp updated_at = 4;
}

// The response message definition, return status of updation.
message UpdateMemberResponse {
 bool is_member_updated = 1;
}
